ksp_plugin_adapter/optional_marshaler.cs

```diff
@@ -35,16 +35,14 @@ internal class OptionalMarshaler<T> : ICustomMarshaler where T : struct {
       throw Log.Fatal("The runtime returns null for null objects");
     }
     T value;
-    var value_if_boxed = managed_object as T?;
-    if (value_if_boxed != null) {
-      value = value_if_boxed.Value;
+    if (managed_object is T value_if_boxed) {
```

Comment:
Fancy!

Reply:
Indeed, it does that with a single click.  I didn't even know that this existed.